the-collab-lab/tcl-34-smart-shopping-list
# Firebase configuration

## Description
This PR includes **_two components_** , **_one custom hook_** and **_two functions_**:

**Components:**

1. `CreateItem`: it is a button that creates a new item by a given default name for testing purposes
2. `ListItem`: displays a `<li>` with the items created. This component fetches Firebase API to retrieve them

**Custom Hook:**

1. `useProducts`: it is responsible for fetch data from Firebase and listen data changes

**Functions**

1. `helpers/firebaseHelpers.js`: we have a function that receives the results obtained in Firebase and parses the data, leaving them ready to be displayed on screen
2. `utils/firebaseUtils.js`: we have a function responsible for creating products in the Firebase

## Related Issue

Closes #1 

## Acceptance Criteria

- [x] Se han añadido: firebase & _react-firebase-hooks*_ como dependencias del proyecto.
- [x] Realice un cambio en la base de datos de Firestore y este aparecerá en la aplicación.
- [x] Realice un cambio en el app y este aparecerá en la base de datos de Firestore.

*It was recommended not to use _react-firebase-hooks_ as it breaks code, that is why we added custom hook.
## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

## Gif

![collab-gif](https://user-images.githubusercontent.com/37032132/136102894-90886141-d135-49bd-9919-ffa728008371.gif)

## Testing Steps / QA Criteria

1. From your terminal, pull down this branch with `git pull origin il-yu-config-firebase` and check that branch out with `git checkout il-yu-config-firebase`
2. Then `npm install` the newly added dependencies locally and `npm start` to launch the app.
3. Click the button to add a new item, it gives 'A new item' name by default. Make sure it adds it to the list and make sure that item appears in the items collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-34-smart-shopping-list/firestore/data/~2Fproducts).
